File: internal/module/tester.go
package module

import (
	"context"

	"github.com/grcengineering/ocean/internal/evidence"
)

// Tester performs active control verification by interacting with target
// systems. Testers produce evidence at the "active_verification" confidence
// level and must declare their safety classification and cleanup procedures.
type Tester interface {
	Module
	SafetyClass() SafetyClassification
	EnvironmentScope() EnvironmentScope
	PreFlightChecks() []string
	CleanupProcedures() []string
	Test(ctx context.Context, config map[string]string) ([]evidence.Evidence, error)
}

File: src/module/tester.rs
use anyhow::Result;
use std::collections::HashMap;

use crate::evidence::Evidence;
use super::{Module, SafetyClassification, EnvironmentScope};

/// Performs active control verification by interacting with target systems.
/// Testers produce evidence at the "active_verification" confidence level and
/// must declare their safety classification and cleanup procedures.
pub trait Tester: Module {
    fn safety_class(&self) -> SafetyClassification;
    fn environment_scope(&self) -> EnvironmentScope;
    fn pre_flight_checks(&self) -> Vec<String>;
    fn cleanup_procedures(&self) -> Vec<String>;
    fn test(&self, config: &HashMap<String, String>) -> Result<Vec<Evidence>>;
}
